crates/pallet-domains/src/weights.rs

```diff
@@ -32,509 +32,887 @@ use core::marker::PhantomData;
-	/// Proof Skipped: Domains ExecutionInbox (max_values: None, max_size: None, mode: Measured)
-	/// Storage: Domains SuccessfulBundles (r:1 w:1)
-	/// Proof Skipped: Domains SuccessfulBundles (max_values: None, max_size: None, mode: Measured)
-	/// Storage: Domains HeadDomainNumber (r:1 w:1)
-	/// Proof Skipped: Domains HeadDomainNumber (max_values: None, max_size: None, mode: Measured)
-	/// Storage: Domains DomainTxRangeState (r:1 w:1)
-	/// Proof Skipped: Domains DomainTxRangeState (max_values: None, max_size: None, mode: Measured)
-	/// Storage: Domains InboxedBundle (r:0 w:2)
-	/// Proof Skipped: Domains InboxedBundle (max_values: None, max_size: None, mode: Measured)
+	/// Storage: `Domains::HeadReceiptNumber` (r:1 w:1)
+	/// Proof: `Domains::HeadReceiptNumber` (`max_values`: None, `max_size`: None, mode: `Measured`)
+	/// Storage: `Domains::LatestConfirmedDomainBlock` (r:1 w:1)
+	/// Proof: `Domains::LatestConfirmedDomainBlock` (`max_values`: None, `max_size`: None, mode: `Measured`)
+	/// Storage: `Domains::BlockTree` (r:2 w:2)
+	/// Proof: `Domains::BlockTree` (`max_values`: None, `max_size`: None, mode: `Measured`)
+	/// Storage: `Domains::BlockTreeNodes` (r:1 w:2)
+	/// Proof: `Domains::BlockTreeNodes` (`max_values`: None, `max_size`: None, mode: `Measured`)
+	/// Storage: `Domains::LatestSubmittedER` (r:1 w:0)
+	/// Proof: `Domains::LatestSubmittedER` (`max_values`: None, `max_size`: None, mode: `Measured`)
+	/// Storage: `Domains::ExecutionInbox` (r:2 w:1)
+	/// Proof: `Domains::ExecutionInbox` (`max_values`: None, `max_size`: None, mode: `Measured`)
+	/// Storage: `Transporter::DomainBalances` (r:1 w:1)
+	/// Proof: `Transporter::DomainBalances` (`max_values`: None, `max_size`: None, mode: `Measured`)
+	/// Storage: `Domains::SuccessfulBundles` (r:1 w:1)
+	/// Proof: `Domains::SuccessfulBundles` (`max_values`: None, `max_size`: None, mode: `Measured`)
+	/// Storage: `Domains::HeadDomainNumber` (r:1 w:1)
+	/// Proof: `Domains::HeadDomainNumber` (`max_values`: None, `max_size`: None, mode: `Measured`)
+	/// Storage: `Domains::InboxedBundleAuthor` (r:0 w:1)
+	/// Proof: `Domains::InboxedBundleAuthor` (`max_values`: None, `max_size`: None, mode: `Measured`)
+	/// Storage: `Domains::HeadReceiptExtended` (r:0 w:1)
+	/// Proof: `Domains::HeadReceiptExtended` (`max_values`: None, `max_size`: None, mode: `Measured`)
+	/// Storage: `Domains::ConsensusBlockHash` (r:0 w:1)
+	/// Proof: `Domains::ConsensusBlockHash` (`max_values`: None, `max_size`: None, mode: `Measured`)
 	fn submit_bundle() -> Weight {
 		// Proof Size summary in bytes:
-		//  Measured:  `3952`
-		//  Estimated: `12367`
-		// Minimum execution time: 102_000_000 picoseconds.
-		Weight::from_parts(118_000_000, 12367)
-			.saturating_add(T::DbWeight::get().reads(10_u64))
-			.saturating_add(T::DbWeight::get().writes(12_u64))
+		//  Measured:  `1609`
+		//  Estimated: `7549`
+		// Minimum execution time: 82_000_000 picoseconds.
+		Weight::from_parts(86_000_000, 7549)
```

Comment:
Are these worst case estimates? or is this fixed cost for farmer because the rest of execution happens on operator side?

Reply:
>Are these worst case estimates?

Bad naming again 🤦‍♂️ it should be `basic_submit_bundle` it is the worst case estimation for the basic flow of `submit_bundle` extrinsic, it covers:
- Add the ER into the block tree
- Prune the oldest (confirmed) ER
- Increase `HeadReciptNumber` and `HeadDomainNumber`, and update `SuccessfulBundles`, `ExecutionInbox`, `InboxedBundleAuthor`
- etc

also the database read/write weight in the next 2 line should be counted.

For other work that will be done within the `submit_bundle` extrinsic like
- Prune bad ER and slash submitter (as lazily process bad ER descendant)
- Confirm ER, refund storage fee, reward submitter and slash invalid bundle submitter
- Epoch transition

They are benchmarked separately, they are counted for the max weight of `submit_bundle` but they are only added to the actual weight if they did happen. See [`fn max_submit_bundle_weight`](https://github.com/subspace/subspace/pull/2592/files/ebf5ff7b20ba9beddc28147522946b0528cfe268#diff-9f80450addc12cd3b681275c6acb663003edc27c2d8eeeb1f26147df9f8e5bbe)

>or is this fixed cost for farmer because the rest of execution happens on operator side?

Right, all of the weight of `pallet-domains` is for the farmer, the weight of executing the domain tx is not counted here since the farmer won't execute the domain tx.

Reply:
Almost, I think we also need to take the weight for `confirm_domain_block` into count, which will happen on every consensus block that contains a bundle after the first challenge period is passed. But it should only happen once in a consensus block, i.e. if the consensus block contains multiple bundles only the first bundle has `confirm_domain_block`.

For other things, `handle_bad_receipt` only happen when there is FP, and `max_staking_epoch_transition` happen once 100 domain blocks (`StakeEpochDuration`).